app/deepseg.cc

```diff
@@ -464,7 +454,13 @@ int main(int argc, char* argv[]) {
 
 	cv::Mat mask(height, width, CV_8U);
 	cv::Mat raw;
-	CalcMask ai(modelname, threads, width, height);
+	CalcMask *ai;
+	try {
+		ai = new CalcMask(modelname, threads, width, height);
```

Comment:
Use `std::unique_ptr<>`/`std::make_unique` or `std::shared_ptr`/`std::make_shared`. Also saves on manual resource management on exit.

Reply:
And why change `ai` from stack object to pointer?

Reply:
Because of the exception in the constructor.

Reply:
This shouldn't make a difference, when you stretch the `try`/`catch` over the whole lifetime of `ai` required by the code below. Also inside main, you can just `return 1;` whcih takes care of all the automatic dtor calls required.

Reply:
The lifetime of `ai` covers the whole main loop.
This seems like a large block for a try.
Do you think, its better to wrap the main function itself with a try instead?

Reply:
You mean like:

```cpp
int main() try {
    // Code
} catch(const char* msg) {
    fprintf(stderr, "Error: %s\n", msg);
    return 1;
}
```

Fine by me … ;-)

Reply:
yes, fixed it.

Reply:
Looks resolved :smile: